### PR TITLE
Update scalar docs so that it's a bit easier to read

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

Update the Scalar documentation to be a bit easier to read.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: docs/02_types/scalars.md [hunk 1]
```diff
@@ -23,31 +25,21 @@ Scalar types represent concrete values at the leaves of a query. For example in
...
+* `Boolean`, true or false, maps to Python’s `bool`
+* `ID`, a specialised `String` for representing unique object identifiers
+* `Date`, an ISO 8601 encoded [date](https://docs.python.org/3/library/datetime.html#date-objects)
+* `DateTime`, an ISO 8601 encoded [datetime](https://docs.python.org/3/library/datetime.html#datetime-objects)
+* `Time`, an ISO 8601 encoded [time](https://docs.python.org/3/library/datetime.html#time-objects)
+* `Decimal`, a [Decimal](https://docs.python.org/3/library/decimal.html#decimal.Decimal) value serialized as a string
+* `UUID`, a [UUID](https://docs.python.org/3/library/uuid.html#uuid.UUID) value serialized as a string
 
-Strawberry also provides some built in scalars for common Python datatypes:
+Fields can return built-in scalars by using the Python equivilant:
```
Comment (patrick91) [inline docs/02_types/scalars.md:42]:
```suggestion
Fields can return built-in scalars by using the Python equivalent:
```

File: docs/02_types/scalars.md [hunk 2]
```diff
@@ -23,31 +25,21 @@ Scalar types represent concrete values at the leaves of a query. For example in
 }
 ```
 
-Strawberry will automatically map the Python built in types to the corresponding
-GraphQL type:
+There are several built-in scalars, and you can define custom scalars too.
+([Enums](/docs/types/enums) are also leaf values.) The built in scalars are:
```
Comment (patrick91) [inline docs/02_types/scalars.md:29]: `leaf values` would it be useful to explain what a leaf value is?

**Review by patrick91 — COMMENTED** [review]
Looks good, added two minor notes :)

File: docs/01_general/01_schema-basics.md [hunk 3]
```diff
@@ -101,24 +101,24 @@ GraphQL supports a few different types:
...
+  for a cache. Serialized as string and available as `strawberry.ID(“value”)`
 
-> **NOTE:** strawberry also includes support for date, time and datetime
+> **NOTE:** Strawberry also includes support for date, time and datetime
 > objects, they are not officially included with the GraphQL spec, but they are
 > usually needed in most servers. They are serialized as ISO-8601.
 
 <!--alex ignore-->
 
 These primitives work for the majority for the uses cases, but you can also
```
Comment (BryceBeagle) [inline docs/01_general/01_schema-basics.md:120]:
```suggestion
These primitives work for the majority of use cases, but you can also
```

File: docs/02_types/scalars.md [hunk 4]
```diff
@@ -23,31 +25,21 @@ Scalar types represent concrete values at the leaves of a query. For example in
...
-
-## Built in scalars
+* `String`, maps to Python’s `str`
+* `Int`, a signed 32-bit integer, maps to Python’s `int`
+* `Float`, a signed double-precision floating-point value, maps to Python’s `float`
+* `Boolean`, true or false, maps to Python’s `bool`
+* `ID`, a specialised `String` for representing unique object identifiers
+* `Date`, an ISO 8601 encoded [date](https://docs.python.org/3/library/datetime.html#date-objects)
+* `DateTime`, an ISO 8601 encoded [datetime](https://docs.python.org/3/library/datetime.html#datetime-objects)
+* `Time`, an ISO 8601 encoded [time](https://docs.python.org/3/library/datetime.html#time-objects)
```
Comment (BryceBeagle) [inline docs/02_types/scalars.md:38]: To stay consistent with `01_schema-basics.md` we should use `ISO-8601` here

**Review by BryceBeagle — CHANGES_REQUESTED** [review]
Couple more minor suggestions

File: docs/02_types/scalars.md (hunk 2 above)
Comment (jkimbo) [reply docs/02_types/scalars.md:29]: Yeah it probably would. How would you describe it though?

Comment (patrick91) [reply docs/02_types/scalars.md:29]: No idea, maybe we can leave like it is for now :)

File: docs/01_general/01_schema-basics.md (hunk 3 above)
Comment (BryceBeagle) [reply docs/01_general/01_schema-basics.md]: Almost there. `uses cases` -> `use cases`

Comment (jkimbo) [reply docs/01_general/01_schema-basics.md]: Damn, missed that

**Review by patrick91 — APPROVED** [review]

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**